.github/workflows/dotnet.yml

```diff
@@ -3,7 +3,11 @@
 
 name: .NET build
 
-on: [push, pull_request]
+on:
+  pull_request:
+  push:
+    branches:
+      - master
```

Comment:
this matches the behavior in SpacetimeDB (https://github.com/clockworklabs/SpacetimeDB/blob/master/.github/workflows/ci.yml)

Reply:
I think SpacetimeDB has somewhat different constraints because it's so large / expensive to build, whereas this repo takes less than a minute.

I love being able to run CI on branches even before I make a PR. Could we preserve that?

Reply:
updated to run on `push` events but _not_ `ssynchronize` events